spec/controllers/users/sessions_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Users::SessionsController, type: :controller do
+  before { @request.env['devise.mapping'] = Devise.mappings[:user] }
+
+  describe 'GET #new' do
+    context 'for authenticated user' do
+      before do
+        sign_in create(:user)
```

Comment:
Where is the value of ":user" location, shouldn't there be some .yml file that contains the data?

Reply:
and i think that there should be some filtering first in the User model? to ensure that there is no creation of user with invalid email format, too short username, etc.

Reply:
They are defined by factory bot under factories folder